src/pyscipopt/scip.pxd

```diff
@@ -1557,10 +1583,23 @@ cdef extern from "scip/pub_lp.h":
     SCIP_Real SCIPcolGetPrimsol(SCIP_COL* col)
     SCIP_Real SCIPcolGetLb(SCIP_COL* col)
     SCIP_Real SCIPcolGetUb(SCIP_COL* col)
+    int SCIPcolGetNLPNonz(SCIP_COL* col)
+    int SCIPcolGetNNonz(SCIP_COL* col)
+    SCIP_ROW** SCIPcolGetRows(SCIP_COL* col)
+    SCIP_Real* SCIPcolGetVals(SCIP_COL* col)
+    int SCIPcolGetIndex(SCIP_COL* col)
+    SCIP_Real SCIPcolGetObj(SCIP_COL *col)
 
 cdef extern from "scip/scip_tree.h":
     SCIP_RETCODE SCIPgetOpenNodesData(SCIP* scip, SCIP_NODE*** leaves, SCIP_NODE*** children, SCIP_NODE*** siblings, int* nleaves, int* nchildren, int* nsiblings)
 
+cdef extern from "scip/struct_branch.h":
```

Comment:
I don't think that this is needed. Please remove it.

Reply:
This was here because without it, I get a compiler error: `incomplete definition of type 'struct SCIP_Branchrule'` (caused by the callback definition in line 354). Looking into this a bit further, it turns out that it is sufficient to just include the file `scip/struct_branch.h`, without importing anything from it.

This seems a bit odd to me. I wonder: is this actually a problem in `type_branch.h` in SCIP?

Reply:
No, I don't think so. Clearly, `type_branch.h` is needed if you include `pub_branch.h`. This allows you to access the API methods related to, e.g., branching rules, but you want to ensure that the user does not have access to the struct of a branching rule.

This is also my problem with this change because PySCIPOpt should not give the user access to some of the structs.